exonum/src/storage/proof_map_index/key.rs

```diff
@@ -465,17 +465,38 @@ impl StorageKey for ProofPath {
 impl PartialOrd for ProofPath {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         if self.start() != other.start() {
-            None
-        } else {
-            for i in self.start()..min(self.end(), other.end()) {
-                let cmp = self.bit(i).cmp(&other.bit(i));
-                if cmp != Ordering::Equal {
-                    return Some(cmp);
-                }
+            return None;
+        }
+        // NB: This check can be moved to "real" code; the method below does not work
```

Comment:
The new implementation improves performance of the proof verification (measured by the corresponding bench) by ~35%, making it almost precisely equal to the one before changing ordering of bits in `ProofPath`. As a downside it's less understandable and I'm not completely sure about its correctness (it *is* tested, incl. some fuzzing).